### PR TITLE
Iptv 2982 melanite

### PR DESCRIPTION
**Pull request by omair100**

### CODE REVIEW

File: test/integration/identify-spec.js [hunk 1]
```diff
@@ -4,14 +4,26 @@ const app = require('../../src/app')
...
+  })
+  describe('if given an invalid user agent', () => {
+    it('Returns an error message', (done) => {
+      request.get('/identify/ua/some-fake-user-agent/json')
         .end((err, res) => {
           if (err) throw err
           expect(res.status).toBe(200)
-          expect(res.body.brand).toBeDefined()
-          expect(res.body.model).toBeDefined()
+          expect(res.body.brand).toEqual('generic')
```
Comment (adamjc) [inline test/integration/identify-spec.js]: Is this what we're saying is an eror message?

File: src/middleware/device-identification/device-data.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  if (devices) {
+    req.deviceData = devices
+    return next()
+  } else {
+    return makeRequest()
+      .then(response => {
+        devices = response.body
+        req.deviceData = devices
+        return next()
+      })
```
Comment (swarnv01) [inline src/middleware/device-identification/device-data.js]: Can we catch error here ?

Comment (adamjc) [inline src/middleware/device-identification/device-data.js]: Good point -- worth checking the response to see if it's valid

File: test/integration/identify-spec.js (hunk 1 above)
Comment (AlexChesters) [inline test/integration/identify-spec.js]: This question is open to the team :)

Me and @omair100 don't have any particularly strong opinions as to what we should return here (for either the status code or the response).

Comment (swarnv01) [inline test/integration/identify-spec.js]: I believe we should return as same as it returns with Demi

File: src/middleware/device-identification/user-agent.js [hunk 3]
```diff
@@ -1,5 +1,10 @@
+const melanite = require('melanite')
+
+const request = require('../../common/request-promise')
+
 function identifyDeviceByUserAgent (req, res) {
-  return res.json({brand: 'brand', model: 'model'})
+  const match = melanite.match(req.deviceData)
+  return res.json(match(req.params.ua))
```
Comment (swarnv01) [inline src/middleware/device-identification/user-agent.js]: I belive the ua we are getting will be encoded ?

File: src/middleware/device-identification/device-data.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  return request.get('https://connected-tv.files.bbci.co.uk/device-identification-data/data.json')
+}
+
+setInterval(_ => {
+  return makeRequest()
+    .then(response => devices = response.body)
+    .catch(err => {
+      logger.error('Error requesting device data', error)
+    })
+}, 300000);
```
Comment (swarnv01) [inline src/middleware/device-identification/device-data.js]: Are we find with 5 minutes interval ?

File: src/middleware/device-identification/user-agent.js [hunk 5]
```diff
@@ -1,5 +1,9 @@
+const melanite = require('melanite')
+
 function identifyDeviceByUserAgent (req, res) {
-  return res.json({brand: 'brand', model: 'model'})
+  const match = melanite.match(req.deviceData)
+  const device = match(req.params.ua)
+  return device ? res.json(device) : res.status(404).json({brand: null, model: null})
```
Comment (swarnv01) [inline src/middleware/device-identification/user-agent.js]: Can we also please log if there is an error or miss match UA

File: src/middleware/device-identification/device-data.js (hunk 2 above)
Comment (AlexChesters) [inline src/middleware/device-identification/device-data.js]: We now catch this error

File: test/integration/identify-spec.js [hunk 6]
```diff
@@ -4,14 +4,26 @@ const app = require('../../src/app')
 const request = req(app.listen())
 
 describe('Identify', () => {
-  describe('if given a user agent', () => {
-    it('Returns a brand and model key', (done) => {
-      request.get('/identify/ua/some-user-agent/json')
+  describe('if given a valid user agent', () => {
+    it('Returns the correct device', (done) => {
+      request.get('/identify/ua/Google%20Chrome%20device/json')
```
Comment (johnbeech) [inline test/integration/identify-spec.js]: Does the endpoint still respond correctly if a the user agent contains a `/` character?

Comment (AlexChesters) [inline test/integration/identify-spec.js]: Nope, we've raised this in Slack - I'll document it on the ticket

**Pull request merged**